nilearn/maskers/tests/test_multi_nifti_labels_masker.py

```diff
@@ -0,0 +1,321 @@
+
+    # Should work with 4D + 1D input too (also test fit_transform)
+    signals_input = [fmri11_img, fmri11_img]
+    signals11_list = masker11.fit_transform(signals_input)
+    assert len(signals11_list) == len(signals_input)
+    for signals in signals11_list:
+        assert signals.shape == (length, n_regions)
+
+    # NiftiLabelsMasker should not work with 4D + 1D input
+    signals_input = [fmri11_img, fmri11_img]
+    masker11 = NiftiLabelsMasker(labels11_img, resampling_target=None)
+    with pytest.raises(DimensionError, match="incompatible dimensionality"):
+        masker11.fit_transform(signals_input)
+
+    # Test all kinds of mismatch between shapes and between affines
+    masker11 = MultiNiftiLabelsMasker(labels11_img, resampling_target=None)
+    masker11.fit()
+    pytest.raises(ValueError, masker11.transform, fmri12_img)
+    pytest.raises(ValueError, masker11.transform, fmri21_img)
+
+    masker11 = MultiNiftiLabelsMasker(labels11_img, mask_img=mask12_img,
+                                      resampling_target=None)
+    pytest.raises(ValueError, masker11.fit)
+
+    masker11 = MultiNiftiLabelsMasker(labels11_img, mask_img=mask21_img,
+                                      resampling_target=None)
+    pytest.raises(ValueError, masker11.fit)
+
+    # Transform, with smoothing (smoke test)
+    masker11 = MultiNiftiLabelsMasker(labels11_img, smoothing_fwhm=3,
+                                      resampling_target=None)
+    signals11_list = masker11.fit().transform(signals_input)
+    for signals in signals11_list:
+        assert signals.shape == (length, n_regions)
+
+    masker11 = MultiNiftiLabelsMasker(labels11_img, smoothing_fwhm=3,
+                                      resampling_target=None)
+    signals11_list = masker11.fit_transform(signals_input)
+    for signals in signals11_list:
+        assert signals.shape == (length, n_regions)
+
+        with pytest.raises(ValueError, match='has not been fitted. '):
+            MultiNiftiLabelsMasker(labels11_img).inverse_transform(signals)
+
+    # Call inverse transform (smoke test)
+    for signals in signals11_list:
+        fmri11_img_r = masker11.inverse_transform(signals)
+        assert fmri11_img_r.shape == fmri11_img.shape
+        np.testing.assert_almost_equal(fmri11_img_r.affine, fmri11_img.affine)
+
+
+def test_multi_nifti_labels_masker_reduction_strategies():
+    """Tests:
+    1. whether the usage of different reduction strategies work.
+    2. whether unrecognised strategies raise a ValueError
+    3. whether the default option is backwards compatible (calls "mean")
+    """
+    test_values = [-2., -1., 0., 1., 2]
+
+    img_data = np.array([[test_values,
+                          test_values]])
+
+    labels_data = np.array([[[0, 0, 0, 0, 0],
+                             [1, 1, 1, 1, 1]]], dtype=np.int8)
+
+    affine = np.eye(4)
+    img = nibabel.Nifti1Image(img_data, affine)
+    labels = nibabel.Nifti1Image(labels_data, affine)
```

Comment:
I think you can refactor this test by:

- using a fixture for the data generation step
- parametrize for the strategy